.chloggen/ottl-fix-ismatch.yaml

```diff
@@ -0,0 +1,20 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: pkg/ottl
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix issue where IsMatch returned an error if the target val was nil
+
+# One or more tracking issues related to the change
+issues: [17572]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  - Affected components
```

Comment:
What's the usual approach with changes to libraries like this? Should each impacted component get its own changelog entry? I'm good with doing it like this, but could also see the other side of things.

Reply:
I don't think we have an established pattern. [I did this once before in the 0.65.0 release](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CHANGELOG.md#v0650).